src/test/java/org/broadinstitute/hellbender/utils/samples/MendelianViolationUnitTest.java

```diff
@@ -146,15 +148,25 @@ public void testIsViolationFromSamples(final Genotype gMom, final Genotype gDad,
         alleles.addAll(gChild.getAlleles());
         alleles.add(refAllele);
         alleles.remove(noCallAllele);
-        final VariantContext vc = new VariantContextBuilder("test", "20", 10, 10, alleles).genotypes(context).make();
 
-        final MendelianViolation mv = new MendelianViolation(0, false, false);
-        final boolean actual = mv.isViolation(sMom, sDad, sChild, vc);
-        Assert.assertEquals(expected, actual);
-        Assert.assertEquals(gMom.isHomRef() && gDad.isHomRef() && gChild.isHet(), mv.getParentsRefRefChildHet() > 0);
-
-        final boolean actualFiltered = new MendelianViolation(GQ30+1, false, false).isViolation(sMom, sDad, sChild, vc);
-        Assert.assertEquals(false, actualFiltered);
+        return new VariantContextBuilder("test", "20", 10, 10, alleles).genotypes(context).make();
+    }
 
+    private void validateMV(final Genotype gMom, final Genotype gDad, final Genotype gChild, final boolean expected, final int minGenotypeQuality) {
+        final VariantContext vc = getVC(gMom, gDad, gChild);
+        final MendelianViolation mv = new MendelianViolation(minGenotypeQuality, false, false);
+        final boolean actual = mv.isViolation(sMom, sDad, sChild, vc);
+        final boolean isNoCall = (!gMom.isCalled() && !gDad.isCalled() || !gChild.isCalled());
+        final boolean isLowQual = minGenotypeQuality > 0 && (gMom.getGQ() < minGenotypeQuality || gDad.getGQ() < minGenotypeQuality || gChild.getGQ() < minGenotypeQuality);
+        final boolean expectedWithQual = !isLowQual && expected;
+
+        Assert.assertEquals(expectedWithQual, actual);
+        Assert.assertEquals(!isLowQual && (gMom.isHomRef() && gDad.isHomRef() && gChild.isHet()), mv.getParentsRefRefChildHet() > 0);
+        Assert.assertEquals(!isLowQual && !isNoCall, mv.getFamilyCalledCount() > 0);
+
+        Assert.assertEquals(isNoCall, mv.getFamilyNoCallCount() > 0);
+        Assert.assertEquals(!isNoCall && isLowQual, mv.getFamilyLowQualsCount() > 0);
+        Assert.assertEquals(expectedWithQual, mv.getViolationsCount() > 0);
+        Assert.assertEquals((isLowQual || isNoCall || (gMom.isHomRef() && gDad.isHomRef() && gChild.isHomRef())), mv.getVarFamilyCalledCount() == 0);
```

Comment:
We generally try not to gang up multiple test/asserts like this in the same method if we can avoid it. Can you separate the tests for getParentsRefRefChildHet , getFamilyNoCallCount , getFamilyLowQualsCount, getVarFamilyCalledCount into separate test methods, and then they can share a minimal utility method to create the vc. Also, the assert methods generally take arguments in the order (actual, expected).

Reply:
done